TRANSCEIVER-5.2: Restructured using functions from internal library

snaragund opened the pull request:
- Defined `TunableParamsTest` which validates optical-channel telemetry. 
- Added 3 more conditions in `isOpticalChannelStreamReady` func. 
- Added required deviations in metadata.textproto.
- Used `tunableParamsTest` func to restructure zrp_tunable_parameters_test.

"This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia's intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind."

snaragund committed:
TRANSCEIVER-5.2: Restructured using itemized functions from internal library
-Defined `TunableParamsTest` which validates optical-channel telemetry.
-Added 3 more conditions in `isOpticalChannelStreamReady` func.
-Added required deviations in metadata.textproto.
-Used `tunableParamsTest` func to restructure zrp_tunable_parameters_test.

"This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia's intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind."

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge remote-tracking branch 'upstream/main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund commented:
@dplore PR has been approved by both @rezachit @rohit-rp can this be merged?

dplore commented:
> @dplore PR has been approved by both @rezachit @rohit-rp can this be merged?

Please fix static checks
https://github.com/openconfig/featureprofiles/actions/runs/21852422973/job/63061952704?pr=4908

Please provide link to test results.  Can either use a github gist or privately.

snaragund committed:
Merge remote-tracking branch 'upstream/main' into SND-TRANSCEIVER5dot2
Fix merge conflict

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Fixing static analysis error.

"This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia's intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind."

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund commented:
@dplore changes are approved could you please merge changes?

snaragund commented:
@rezachit @dplore @rohit-rp could you please merge if no further comments?

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund committed:
Merge branch 'main' into SND-TRANSCEIVER5dot2

snaragund commented:
@rezachit have added some more improvements please review the changes.

navaneethyv commented:
/gemini review

alokmtri-g commented:
/fptest physical
